### DISCUSSION TITLE
ITEM 6 APIMLibre

### DISCUSSION THREAD

**Discussion by bmangia** [dds-jv/foro#144] [TPA - 2020] [answered]
Buenas

Consulta con respecto al item 6:

_**La información de países, provincias y ciudades se deben obtener mediante la utilización de la API de Mercado Libre. 
De igual forma, se busca incorporar el concepto de moneda a todo valor monetario; la información de las monedas se debe obtener también mediante el API antedicha.**_

No me queda bien claro la utlidad que le vamos a dar a la API, mas alla de obtener la moneda segun el pais.

En el momento que creo un proveedor, ¿Tengo que restringirlo a que sus atributos: ciudad provincia pais sean solo los que levanto de la API? ¿Va por ese lado el uso?

**Comment by roli93** [comment]
Buenas @bmangia! 

En realidad es un requerimiento similar al punto 4, pero con una vuelta de tuerca. En el punto 4 necesitás tener acceso de alguna forma a "todos los presupuestos" para validarlos. Esos presupuestos estarán en el sistema porque antes alguien los fue cargando

En el punto 6, la idea es que (más allá de cómo vaya a ser la UI a futuro) cuando un usuario quiera crear un proveedor, los datos postales que le cargue tendrán que haber salido de algún lado donde estén "todos los datos postales posibles". En este caso, ese universo de datos posibles no es precargado por un administrador, sino que el sistema los tomará automáticamente de MercadoLibre

**Comment by resLotus** [comment]
Sumándome a la duda, cuando decimos que es similar al requerimiento del punto 4, esto significa que también debe actuar de forma a sincrónica? 
Por ejemplo yo tendría que poder cargar los datos de un código postal y recién chequear que dichos datos sean validos cuando agrego el código postal al proveedor?

**Comment by roli93** [comment]
Nunca tenes que 'chequear' ni validar ningun dato. Porque los datos que ingrese el usuario van a ser de entre las opciones que vos le des, que nunca van a estar mal (confiar en el adentro). Esas opciones tenés que sacarlas de algún lado. 

En lo único que se parece al punto 4 es que tenés que tener un lugar donde estén **todos** (por ej, en el punto 4 necesitás un _repositorio de usuarios_, donde estan **todos** los usuarios). No hay asincornismo en el punto 6.

Acá es una idea similar: necesitás un objeto que te provea, por ejemplo, **todas** las ciudades (y países, provincias, etc). Ese objeto las ciudades las va a sacar de la API de ML. Las puede pedir cada vez, o almacenarlas en memoria. Eso es una decisión que tomarán ustedes y justificarán, charlándola con su ayudante.

**Comment by resLotus** [comment]
Perfecto, muchas gracias por la respuesta quedo más claro

**Comment by roli93** [comment]
@bmangia @resLotus Les quedó alguna duda más?

**Comment by resLotus** [comment, marked as answer]
No, muchas gracias Roli, nos vemos el viernes
